### PR TITLE
Fix Azure Storage `RunAsEmulator`

### LINKED ISSUE #5078: Azure Storage component doesn't work with Azure Storage RunAsEmulator
### Repro Steps
In your AppHost:

```csharp
var builder = DistributedApplication.CreateBuilder(args);

var blobs = builder.AddAzureStorage("Storage")
    // Use the Azurite storage emulator for local development
    .RunAsEmulator()
    .AddBlobs("BlobConnection");

builder.AddProject<Projects.AspireApp21_Web>("webfrontend")
    .WithExternalHttpEndpoints()
    .WithReference(blobs);
```

In your webfrontend project:

```csproj
var builder = WebApplication.CreateBuilder(args);

// Add service defaults & Aspire components.
builder.AddServiceDefaults();

// add this line
builder.AddAzureBlobClient("BlobConnection");
```

F5

hit the `webfrontend/health` endpoint

### Expected results

Healthy

### Actual results

Unhealthy

### Logs

```
      client assembly: Azure.Storage.Blobs
2024-07-25T12:37:42.5405046 warn: Azure.Core[8]
      Error response [ab557070-d043-4751-975d-10a24ee30e85] 400 The API version 2024-08-04 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error.  (00.0s)
      Server:Azurite-Blob/3.30.0
      x-ms-error-code:InvalidHeaderValue
      x-ms-request-id:d221c3a6-dc25-4191-ab4f-0da95f77d798
      Date:Thu, 25 Jul 2024 17:37:42 GMT
      Connection:keep-alive
      Keep-Alive:REDACTED
      Transfer-Encoding:chunked
      Content-Type:application/xml
      
2024-07-25T12:37:42.5596003 fail: Microsoft.Extensions.Diagnostics.HealthChecks.DefaultHealthCheckService[103]
      Health check Azure_BlobServiceClient with status Unhealthy completed after 69.5226ms with message '(null)'
      Azure.RequestFailedException: The API version 2024-08-04 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error. 
RequestId:d221c3a6-dc25-4191-ab4f-0da95f77d798
Time:2024-07-25T17:37:42.527Z
      Status: 400 (The API version 2024-08-04 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error. )
      ErrorCode: InvalidHeaderValue
```

If we update to image tag `3.31.0` here, the problem goes away:

https://github.com/dotnet/aspire/blob/e0552c0bbf3330c2694e72961a97941a5773e139/src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs#L114-L119

### PR DESCRIPTION
**Pull request by radical**
And add a test for the same.

Fixes https://github.com/dotnet/aspire/issues/5078.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5084)

### CODE REVIEW

File: tests/Aspire.Hosting.Azure.Tests/Aspire.Hosting.Azure.Tests.csproj [hunk 1]
```diff
@@ -23,6 +23,8 @@
     <ProjectReference Include="..\..\src\Aspire.Hosting.Azure.AppConfiguration\Aspire.Hosting.Azure.AppConfiguration.csproj" />
     <ProjectReference Include="..\Aspire.Hosting.Tests\Aspire.Hosting.Tests.csproj" />
 
+    <ProjectReference Include="..\..\src/Components/Aspire.Azure.Storage.Blobs/Aspire.Azure.Storage.Blobs.csproj" />
```
Comment (mitchdenny) [inline tests/Aspire.Hosting.Azure.Tests/Aspire.Hosting.Azure.Tests.csproj:26]: Nit: inconsistent slashes

**Review by mitchdenny — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks, @radical!

File: tests/Aspire.Hosting.Azure.Tests/AzureStorageEmulatorFunctionalTests.cs [hunk 2]
```diff
@@ -27,20 +25,20 @@ public async Task VerifyAzureStorageEmulatorResource()
...
-        Assert.NotNull(accountInfo);
-        Assert.Equal(AccountKind.StorageV2, accountInfo.AccountKind);
-        Assert.Equal(SkuName.StandardRagrs, accountInfo.SkuName);
+        var blobContainer = (await serviceClient.CreateBlobContainerAsync("container")).Value;
+        var blobClient = blobContainer.GetBlobClient("testKey");
+
+        await blobClient.UploadAsync(BinaryData.FromString("testValue"));
+
+        var downloadResult = (await blobClient.DownloadContentAsync()).Value;
+        Assert.Equal("testValue", downloadResult.Content.ToString());
```
Comment (radical) [inline tests/Aspire.Hosting.Azure.Tests/AzureStorageEmulatorFunctionalTests.cs:41]: better test :) 👍

**Pull request merged**